arbor/mc_cell_group.cpp

```diff
@@ -135,21 +85,54 @@ void mc_cell_group::advance(epoch ep, time_type dt, const event_lane_subrange& e
 
     PE(advance_eventsetup);
     staged_events_.clear();
+
+    cell_size_type ev_begin = 0, ev_mid = 0, ev_end = 0;
```

Comment:
Being at the interface between the public 'cell_ABC' types and the back-end `fvm_XYZ` types means that there will be some compromises in type choices here. For indexes into the big list of staged events, it might be better to go with `fvm_index_type`.

A comment at the beginning of `advance()` clarifying what set of types is used for which information could be helpful, given the ambiguity here.

As an aside, the main reason why `fvm_index_type` is an `int` rather than `unsigned` is that otherwise we'd have to extend the vectorization library to support `unsigned` in an addition to `int`, and that looked like way too much work!